### PR TITLE
Adds ability to set Elastic Search connection parameters via helm chart

### PR DESCRIPTION
**Pull request by mastermanu**

### CODE REVIEW

File: templates/server-deployment.yaml [hunk 1]
```diff
@@ -72,11 +72,17 @@ spec:
...
+              value: "{{ $.Values.elasticsearch.host }}"
             - name: ES_PORT
               value: "{{ $.Values.elasticsearch.port }}"
             - name: ES_SCHEME
               value: "{{ $.Values.elasticsearch.scheme }}"
+            - name: ES_VIS_INDEX
+              value: "{{ $.Values.elasticsearch.visibilityIndex }}"
+            - name: ES_USER
+            - value: "{{ $.Values.elasticsearch.username }}"
+            - name: ES_PWD
```
Comment (mastermanu) [inline templates/server-deployment.yaml:84]: should be able to re-use the temporal-visibility-store-password here below

File: templates/server-configmap.yaml [hunk 2]
```diff
@@ -77,8 +77,10 @@ data:
                 url:
                     scheme: "{{ $.Values.elasticsearch.scheme }}"
                     host: "{{ $.Values.elasticsearch.host }}:{{ $.Values.elasticsearch.port }}"
+                username: "{{ $.Values.elasticsearch.username }}"
```
Comment (underrun) [inline templates/server-configmap.yaml:80]: does this still work if the ES endpoint doesn't require auth?

File: templates/server-configmap.yaml [hunk 3]
```diff
@@ -77,8 +77,10 @@ data:
...
+                password: "{{ $.Values.elasticsearch.password }}"
```
Comment (mastermanu) [inline templates/server-configmap.yaml:81]: should ideally pull this from environment variable

File: templates/server-configmap.yaml (hunk 2 above)
Comment (mastermanu) [reply templates/server-configmap.yaml:80]: Good question. Yes, because the ES client library checks for empty string on username/password

**Review by markmark206 — APPROVED** [review]

**Pull request merged**